framework/src/meshgenerators/TransfiniteMeshGenerator.C

```diff
@@ -0,0 +1,542 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TransfiniteMeshGenerator.h"
+#include "CastUniquePointer.h"
+#include "Conversion.h"
+#include "MooseMeshUtils.h"
+
+#include "libmesh/parsed_function.h"
+#include "libmesh/replicated_mesh.h"
+#include "libmesh/face_quad4.h"
+#include "libmesh/fparser_ad.hh"
+#include "libmesh/elem.h"
+
+registerMooseObject("MooseApp", TransfiniteMeshGenerator);
+
+InputParameters
+TransfiniteMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params += FunctionParserUtils<false>::validParams();
+
+  MooseEnum edge_type("LINE=1 CIRCARC=2 DISCRETE=3 PARSED=4");
+
+  params.addRequiredParam<std::vector<Point>>("corners", "The x,y,z positions of the nodes");
+  // Define edge types
+
+  params.addRequiredParam<MooseEnum>("bottom_type", edge_type, "type of the bottom (y) boundary");
+  params.addRequiredParam<MooseEnum>("top_type", edge_type, "type of the top (y) boundary");
+  params.addRequiredParam<MooseEnum>("left_type", edge_type, "type of the left (x) boundary");
+  params.addRequiredParam<MooseEnum>("right_type", edge_type, "type of the right (x) boundary");
+
+  // We need to know the number of points on opposite sides, and if no other parameter is available
+  // we shall assume them to be equally distributed
+  params.addRequiredParam<unsigned int>("nx",
+                                        "Number of nodes on horizontal edges, including corners");
+  params.addRequiredParam<unsigned int>("ny",
+                                        "Number of Nodes on vertical edges, including corners");
+
+  // each edge has a different parameter according to its type
+  params.addParam<std::string>("bottom_parameter", "", "Bottom side support parameter");
+  params.addParam<std::string>("top_parameter", "", "Top side support parameter");
+  params.addParam<std::string>("left_parameter", "", "Left side support parameter");
+  params.addParam<std::string>("right_parameter", "", "Right side support parameter");
```

Comment:
Even with the latest version of this PR, I'm still seeing the behavior in which the `"left_parameter" ` value governs the behavior on the right edge, and the `"right_parameter"` value sets the behavior on the left edge, regardless of the order in which I provide the points. The created sidesets do follow the expected left and right conventions. It looks to me like the correct variables are being passed `getEdge` on line 158 and 159.

This discrepancy from the naming convention with the parameter arguments can be confusing for the user. Would renaming the parameter strings for left and right here (and lines 96 and 97), with appropriate in-code documentation, be something that you would be willing to consider?